diff --git a/package/openresolv/openresolv.hash b/package/openresolv/openresolv.hash
index c32b856de9..6a014cd75c 100644
--- a/package/openresolv/openresolv.hash
+++ b/package/openresolv/openresolv.hash
@@ -1,3 +1,3 @@
 # Locally calculated
-sha256  36b5bcbe257a940c884f0d74321a47407baabab9e265e38859851c8311f6f0b0  openresolv-3.13.2.tar.xz
+sha256  55d5c9d6f168df59a6dd22afca8c4b15c9149a9c5422c474a8eded35786c5767  openresolv-3.17.3.tar.xz
 sha256  384740bf5e19e7628d20f4dcc22925062fdc6020e9a6f074e567d4d797be57a0  LICENSE
