diff --git a/package/qt5/qt5tools/qt5tools.mk b/package/qt5/qt5tools/qt5tools.mk
index c03d5d6e38..ff65d19052 100644
--- a/package/qt5/qt5tools/qt5tools.mk
+++ b/package/qt5/qt5tools/qt5tools.mk
@@ -30,7 +30,7 @@ QT5TOOLS_INSTALL_STAGING_DIR_$(BR2_PACKAGE_QT5TOOLS_LINGUIST_TOOLS) += \
 ifeq ($(BR2_PACKAGE_QT5TOOLS_QDOC_TOOL),y)
 QT5TOOLS_BUILD_DIRS_y += qdoc
 QT5TOOLS_INSTALL_STAGING_DIR_y += qdoc
-QT5TOOLS_DEPENDENCIES += host-clang
+QT5TOOLS_DEPENDENCIES += reglinux-llvm
 endif
 
 QT5TOOLS_BUILD_DIRS_$(BR2_PACKAGE_QT5TOOLS_PIXELTOOL) += pixeltool
