--- a/lakka.h
+++ b/lakka.h
@@ -18,13 +18,13 @@
 #ifndef __RARCH_LAKKA_H
 #define __RARCH_LAKKA_H
 
-#define LAKKA_SSH_PATH       "/storage/.cache/services/sshd.conf"
-#define LAKKA_SAMBA_PATH     "/storage/.cache/services/samba.conf"
-#define LAKKA_BLUETOOTH_PATH "/storage/.cache/services/bluez.conf"
-#define LAKKA_UPDATE_DIR     "/storage/.update/"
-#define LAKKA_CONNMAN_DIR    "/storage/.cache/connman/"
-#define LAKKA_LOCALAP_PATH   "/storage/.cache/services/localap.conf"
-#define LAKKA_TIMEZONE_PATH  "/storage/.cache/timezone"
+#define LAKKA_SSH_PATH       "/userdata/system/.cache/services/sshd.conf"
+#define LAKKA_SAMBA_PATH     "/userdata/system/.cache/services/samba.conf"
+#define LAKKA_BLUETOOTH_PATH "/userdata/system/.cache/services/bluez.conf"
+#define LAKKA_UPDATE_DIR     "/userdata/system/upgrade/"
+#define LAKKA_CONNMAN_DIR    "/userdata/system/.cache/connman/"
+#define LAKKA_LOCALAP_PATH   "/userdata/system/.cache/services/localap.conf"
+#define LAKKA_TIMEZONE_PATH  "/userdata/system/.cache/timezone"
 
 #define DEFAULT_TIMEZONE "UTC"
 #define TIMEZONE_LENGTH 255
