--- a/es-app/src/guis/GuiMenu.cpp
+++ b/es-app/src/guis/GuiMenu.cpp
@@ -1939,7 +1939,8 @@
 		securityEnabled->setState(SystemConf::getInstance()->get("system.security.enabled") == "1");
 		securityGui->addWithDescription(_("ENFORCE SECURITY"), _("Require a password for accessing the network share."), securityEnabled);
 
-		auto rootpassword = std::make_shared<TextComponent>(mWindow, ApiSystem::getInstance()->getRootPassword(), ThemeData::getMenuTheme()->Text.font, ThemeData::getMenuTheme()->Text.color);
+		auto currentrootpassword = ApiSystem::getInstance()->getRootPassword();
+		auto rootpassword = std::make_shared<TextComponent>(mWindow, &currentrootpassword, ThemeData::getMenuTheme()->Text.font, ThemeData::getMenuTheme()->Text.color);
 		securityGui->addWithLabel(_("ROOT PASSWORD"), rootpassword);
 
 		securityGui->addSaveFunc([this, securityEnabled, s] 
@@ -1951,6 +1952,7 @@
 				SystemConf::getInstance()->set("system.security.enabled", securityEnabled->getState() ? "1" : "0");
 				SystemConf::getInstance()->saveSystemConf();
 				s->setVariable("reboot", true);				
+				currentrootpassword = ApiSystem::getInstance()->getRootPassword();
 			}
 		});
 		mWindow->pushGui(securityGui);
